fecgov/fec-cms
# Use public calendar key for subscriptions only

## Summary (required)
Use `FEC_API_KEY_PUBLIC_CALENDAR` for Calendar subscriptions and use `FEC_API_KEY_PUBLIC` for homepage Event and deadlines links and Calendar filters. 

- Resolves https://github.com/fecgov/fec-accounts/issues/354


## Impacted areas of the application
modified:  fec/fec/static/js/modules/calendar-helpers.js
modified:  fec/fec/static/js/modules/calendar.js
modified:  fec/fec/static/js/pages/calendar-page.js
modified:  fec/fec/static/js/templates/calendar/subscribe.hbs
modified:  fec/fec/tests/js/calendar.js


## Screenshots


![cal-api](https://user-images.githubusercontent.com/5572856/106974817-573c0680-6723-11eb-9831-687c4c3b43e3.jpg)



## How to test
checkout feature/345-use-calendar-key-for-subscriptions-only
`npm run build`
`export FEC_WEB_API_KEY_PUBLIC=12345`
`export FEC_WEB_API_KEY_PUBLIC_CALENDAR=67890`
runserver
Check that the links in subscribe dropdown have  `api_key=67890`
Check in network tab that changes to filters register an api call using `api_key=12345`
Check that clicks from links on homepage Event and deadlines section (except Meetings) register an api call using `api_key=12345` in network tab.

____
